src/server/rdb_save.cc

```diff
@@ -348,6 +348,8 @@ error_code RdbSerializer::SaveListObject(const robj* obj) {
   while (node) {
     DVLOG(3) << "QL node (encoding/container/sz): " << node->encoding << "/" << node->container
              << "/" << node->sz;
+    RETURN_ON_ERR(SaveLen(node->container));
```

Comment:
That was a bug!

Reply:
nice.How come we have not encountered it?
can we add a test in rdb_test for it?

Reply:
not quite sure, the new tests should cover it :)

Reply:
Jesus, because it was redis 7 and quicklist_2. I reverted this, since we want `redis 6` to be able to load our `rdb` files.